lingua_franca/lang/parse_en.py

```diff
@@ -629,10 +634,35 @@ def extract_duration_en(text):
         time_units[unit] = value
         text = re.sub(unit_pattern, '', text)
 
-    text = text.strip()
-    duration = timedelta(**time_units) if any(time_units.values()) else None
+    extended_units = {
+        'months': 0,
+        'years': 0,
+        'decades': 0,
+        'centurys': 0,
+        'millenniums': 0
```

Comment:
Incidental, but it's "millennia" from Latin.

Reply:
both are correct and both should be supported